src/org/rust/lang/annotator/RustAnnotator.kt

```diff
@@ -29,6 +28,26 @@ public class RustAnnotator : Annotator {
             is RustTypeParam -> {
                 addTextAttributes(element, holder, RustColors.TYPE_PARAMETER)
             }
+            is RustPatIdent -> {
+                if (element.bindingMode?.mut != null) {
+                    addTextAttributes(element.identifier, holder, RustColors.MUT_BINDING)
+                }
+            }
+            is RustPathPart -> {
+                val resolved = RustResolveEngine().resolve(element).resolved
+                when (resolved) {
+                    is RustPatField -> {
+                        if (resolved.bindingMode?.mut != null) {
+                            addTextAttributes(element, holder, RustColors.MUT_BINDING)
+                        }
+                    }
+                    is RustPatIdent -> {
+                        if (resolved.bindingMode?.mut != null) {
+                            addTextAttributes(element, holder, RustColors.MUT_BINDING)
+                        }
+                    }
+                }
```

Comment:
This duplication doesn't feel right. Maybe we should extract a PSI node `pat_binding = binding_mode? identifier` from `pat_field` and `pat_ident`. @alexeykudinkin what do you think about it?

At minimum, this can be reduced to something along the lines of

```Kotlin
val isMut = when(resolved) {
  is RustPatField -> resolved.bindingMode?.mut != null,
  is RustPatIdent -> resolved.bindingMode?.mut != null,
  else            -> false
}

if (isMut) {
    addTextAttributes(element, holder, RustColors.MUT_BINDING)
}
```